RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -635,31 +635,37 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
         if (m_do_photon_shapeVar && systemID == systemID_EMB) {
           if (m_do_widthTheta_logE_weights) {
             double w_theta2 = theta2_E_layer[layer+startPositionToFill] / sumWeightLayer[layer+startPositionToFill] - std::pow(theta_E_layer[layer+startPositionToFill] / sumWeightLayer[layer+startPositionToFill], 2);
-            // Very small but negative values caused by computational precision are allowed,
-            // otherwise crash.
-            if (w_theta2 < -1e-9) {
-              error() << "w_theta2 in theta width calculation is negative: " << w_theta2 << endmsg;
-              return StatusCode::FAILURE;
+            // Negative values can happen when noise is on and not filtered
+            // Negative values very close to zero can happen due to numerical precision
+            if (w_theta2 < 0.) {
+              warning() << "w_theta2 in theta width calculation is negative: " << w_theta2 << " , will set theta width to zero (this might happen when noise simulation is on)" <<  endmsg;
```

Comment:
There are cases when the computation is negative when noise is OFF just due to numerical precision, and also for the noise case I think for the user it might be useful to get a least few warnings to make them aware. I have committed a version of the code in which the maximum number of warning statements over the whole execution is configurable and is by default set to 10